src/ui/js/Game.js

```diff
@@ -64,9 +64,21 @@ Game.showLoggedInPage = function() {
 };
 
 // Redraw the page after a previous action succeeded: to do this,
-// clear all activity variables set by the previous invocation
+// clear all activity variables set by the previous invocation except for
+// chat that is currently being actively cached in Game.activity.chat
```

Comment:
At the risk of being picky: i think it would be tidier to make a function for this, like:
```
Game.activity = Game.getActivityAfterApiSuccess();
```

Basically i think futzing around with the chat variable is a little low-level for this function.

My less nitpicky worry is: is this redraw function *only* invoked after chat privacy is changed, or is it invoked after any API function is invoked?  If the latter, won't it retain chat on a successful attack in which you submit your chat?  Did you test that?

Reply:
I tested after a successful attack, and I didn't notice any visible differences because:
- the chat box is not visible after successful attack, and
- if the chat is to be altered, the chat box that becomes visible is filled with the current value of chat anyway

I guess the question is whether we ever really want to clear the chat variable in Activity, considering all the cases in the past where we've really wanted to keep it.

Reply:
Hmm, what if your opponent has to pass?  Anyway, maybe i should just put up a dev site, since it sounds like we might get bogged down in parsing possible options by memory otherwise.  I'll do that when i get a sec.

Reply:
Okay, good pick. I tested with the case where the opponent autopasses and yes, this is one case where we probably don't want to maintain previous chat.

So, based on the fact that there is some sense that the default should still be to clear the chat variable, how would you feel about me using a new Game.activity.chatCache variable to explicitly show that we want to carry the old chat across? Or would you prefer some other approach?

Reply:
Sorry, missed that this ended with a question for me.  As in the variable would be a boolean, and it would be false most of the time, but calling an API like change chat privacy would set it to true?  Seems reasonable to me --- if that's the approach, i would make the name more generic, like `Game.activity.cacheAcrossCall`, in case there's other activity it turns out we want to cache.

Reply:
Okay, sounds good. I'll take another look at this soon.